tools/nl/eval/main.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import json
+import os
+
+from absl import app
+from absl import flags
+from google.cloud import aiplatform
+from google.cloud import aiplatform_v1
+import ndcg
+import numpy as np
+import yaml
+
+PROJECT = 'datcom-website-dev'
+LOCATION = 'us-central1'
+VECTOR_SERACH_ENDPOINT = "302175072.us-central1-496370955550.vdb.vertexai.goog"
+INDEX_ENDPOINT = "projects/496370955550/locations/us-central1/indexEndpoints/8500794985312944128"
+NEIGHBOR_COUNT = 30
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    'model_name', '',
+    'Model name used for eval. Full list can be found in endpoints.yaml')
+
+flags.DEFINE_string(
+    'eval_folder', 'base',
+    'The folder for a eval unit. It should contain a golden.json file')
+
+
+def _load_models():
```

Comment:
nit: maybe comment that this is loading the model mapping/config because at first glance, I thought it was loading the actual models

Reply:
good point. changed to _load_model_endpoints()